### PR TITLE
MODINVOSTO-172 Cannot search for voucher by number after editing voucher number

### PR DESCRIPTION
**Pull request by yuntianhu**
Description

Overview: Performing a search based on voucher number for MG668 (the new voucher number I assigned) produces no results. However, a search for MG68 (the original value) returns the correct record:https://bugfest-mg.int.aws.folio.org/invoice/view/c463c38f-30eb-4a7a-81ec-bda899993a6d?limit=50&offset=0&qindex=voucherNumber&query=MG68
**

Steps to Reproduce:

Log into some FOLIO environment as User X

Create invoice

Approve invoice

Edit voucher number (Add a 1 at beginning of number)

search invoice by voucher number using new number

Expected Results: Invoice is displayed in search result list

Actual Results: No results found for search

Additional Information:
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+UPDATE ${myuniversity}_${mymodule}.vouchers
+SET jsonb = jsonb || jsonb_build_object('voucherNumber', jsonb ->> 'voucherNumber');
+
+UPDATE ${myuniversity}_${mymodule}.invoices AS i
+SET jsonb = jsonb || jsonb_build_object('voucherNumber', v.jsonb ->> 'voucherNumber')
+  FROM ${myuniversity}_${mymodule}.vouchers AS v
+WHERE i.invoiceId = v.invoiceId;
```
Comment (SerhiiNosko) [inline src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql:7]: we also need to update schema.json file to include this fix, next release is 5.9.0

File: src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+UPDATE ${myuniversity}_${mymodule}.vouchers
+SET jsonb = jsonb || jsonb_build_object('voucherNumber', jsonb ->> 'voucherNumber');
```
Comment (SerhiiNosko) [inline src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql:2]: we don't need to update vouchers table, only invoice table to get correct voucher number from voucher table

Comment (yuntianhu) [reply src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql:2]: cool, I will remove this part

File: src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql (hunk 1 above)
Comment (yuntianhu) [reply src/main/resources/templates/db_scripts/data-migration/update_voucherNumver_in_voucher_invoice.sql:7]: sure, I will update the schema.json too, thank you for letting me know : )

File: src/main/resources/templates/db_scripts/schema.json [hunk 3]
```diff
@@ -73,6 +73,11 @@
        "run": "after",
        "snippetPath": "data-migration/5.8.0/update_invoices_fiscal_year_id.sql",
        "fromModuleVersion": "mod-invoice-storage-5.8.0"
+    },
+    {
+      "run": "after",
+      "snippetPath": "data-migration/5.9.0/update_voucherNumber_in_voucher_invoice.sql",
+      "fromModuleVersion": "mod-invoice-storage-5.8.0"
```
Comment (azizbekxm) [inline src/main/resources/templates/db_scripts/schema.json:80]: lets revert back to 5.9.0

**Review by azizbekxm — APPROVED** [review]

**Review by SerhiiNosko — APPROVED** [review]

**Pull request merged**